perceval/components/core_catalog/heralded_cz.py

```diff
@@ -61,21 +59,21 @@ def build(self):
             return self.build_processor(backend=self._opt('backend'))
 
     def build_circuit(self, **kwargs) -> Circuit:
-        # the matrix of this first circuit is the same as the one presented in the reference paper, the difference in the second phase shift - placed on mode 3 instead of mode 1 - is due to a different convention for the beam-splitters (signs inverted in second column).
-        last_modes_cz = (Circuit(4)
-                         .add(0, PS(math.pi), x_grid=1)
-                         .add(3, PS(math.pi), x_grid=1)
-                         .add((1, 2), PERM([1, 0]), x_grid=1)
-                         .add((0, 1), BS.H(theta=self.theta1), x_grid=2)
-                         .add((2, 3), BS.H(theta=self.theta1), x_grid=2)
-                         .add((1, 2), PERM([1, 0]))
-                         .add((0, 1), BS.H(theta=-self.theta1))
-                         .add((2, 3), BS.H(theta=self.theta2)))
+        U = Matrix(
```

Comment:
Can you still use components from perceval with this gate instead of giving an unitary ? The goel here was just to put accurate theta1 or theta2 values. Sorry if it wasn't clear 😬

Reply:
Ok ,it asked for some work. I had to find the exact value of theta1 and theta 2 from the unitary. The reason why the gate was not precise is that the angles 54.74 and 17.63 are rounded up.